README.md

```diff
@@ -37,6 +37,37 @@ within the `modules` directory, and run:
 pip install -e .
 ```
 
+### Noxfile
+
+[Nox](https://nox.thea.codes/en/stable/) is used to automate various development tasks running in isolated python environments.
+The following nox sessions are provided:
+
+- `test`: run tests for all supported python versions
+- `lint`: run `black` and `pylint` for all supported python versions
+- `type_check`: run `mypy` for all supported python versions
+
+To run a session, install the `nox` command (e.g., using pipx) and run:
+
+```shell
+nox -s <session_name>
+```
+
+To save time, reuse the session virtual environment using the `-r` option, i.e. `nox -rs <session_name>`.
+The checks are designed to conform to
+
+### Pre-commit
+
+[Pre-commit](https://pre-commit.com/) can be used to run a configured set of git pre-commit hooks before each commit. This is recommended.
+To set up the pre-commit hooks, install `pre-commit` (e.g., using pipx) and run:
+
+```shell
+pre-commit install
```

Comment:
Does this install the nox and the other requirements as well or does that needs to be done as well?

Reply:
Have you seen this comment?

Reply:
No it doesn't, nox will need to be installed separately

Reply:
Should we explain this somewhere in the readme as well? What do you think?

Reply:
I'll add a note about that in the README